### PR TITLE
dockerfile to build a docker image with SOLVCON.

### PR DESCRIPTION
**Pull request by tai271828**
# Description

This commit adds a dockerfile to build a docker image with SOLVCON. The dockerfile could be used as a part of quick infrastructure to prepare SOLVCON runtime or development environment, e.g. a container-based cluster.

To put and maintain this dockerfile here is better rather than solvcon-gce, because this dockerfile could be run on local machine. For example, a user may run its example on his/her own local machine to verify his/her code or a real SOLVCON case. Please let me know if you think solvcon-gce is a better choice rather than SOLVCON.
# Test Case

Go to where this dockerfile is, and then run the command:

```
docker build -t <your docker repository name>:<your docker tag name> .
```

And then verify it with

```
docker run <your docker repository name>:<your docker tag name> bash -c "source /home/solvcon/miniconda/bin/activate; nosetests /home/solvcon/solvcon/"
```
# Expected Result

docker should report the usual SOLVCON nosetests result like

```
...................................................................................................................................................................................................................................................................................SSS
----------------------------------------------------------------------
Ran 278 tests in 4.759s

OK (SKIP=3)
```

Besides, you should have one more docker image. Confirm this by `docker images`.

### CODE REVIEW

File: contrib/dockerfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+ENV HOME="/home/solvcon"
+ENV PATH="$HOME/miniconda/bin:$PATH"
+ENV DOCKER_SCSRC="$HOME/solvcon"
+RUN cd $HOME; wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh
+RUN bash $HOME/miniconda.sh -b -p $HOME/miniconda
+RUN conda config --set always_yes yes --set changeps1 no
+RUN conda update -q conda
+RUN cd $HOME; git clone https://github.com/solvcon/solvcon.git
+RUN ${DOCKER_SCSRC}/contrib/conda.sh
+RUN cd ${DOCKER_SCSRC}; /bin/bash -c "source activate; python setup.py build_ext --inplace"
```
Comment (tai271828) [inline contrib/dockerfile:57]: This `bash -c` looks not so graceful. docker uses `sh` instead of `bash` by default and no `source` command could be used.

Comment (yungyuc) [reply contrib/dockerfile:57]: I prefer bash to sh.  SOLVCON uses a lot of modern components like the latest Python 3 (I am waiting for 3.6 release) and C++11.  I expect systems support them should also support bash.

File: contrib/dockerfile [hunk 2]
```diff
@@ -0,0 +1,16 @@
+FROM ubuntu:14.04
+MAINTAINER Taihsiang Ho <tai271828@gmail.com>
+RUN apt-get -qq update
+RUN apt-get -qqy install g++ liblapack-dev git wget
+RUN useradd -m solvcon
```
Comment (yungyuc) [inline contrib/dockerfile:43]: I like it.

**Comment by yungyuc** [comment]
Too bad pybind11 fails the build.  ~~I am fixing it.~~  I fixed the build.

**Review by yungyuc — CHANGES_REQUESTED** [review]
Could you organize the comments (test cases and expected results) you put in the PR to the file itself?

**Comment by tai271828** [comment]
Resubmit the PR to add the comments in the dockerfile. Please review it.

**Comment by yungyuc** [comment]
I didn't compare word by word, but it seems you copied everything from the PR description?  The "... commit ..." looks out of place inside the file.  Could you please do some organization / editing?

**Comment by yungyuc** [comment]
And please insert a white space between remark texts and the leading dial mark.

**Comment by tai271828** [comment]
Thanks for the reminder. I fixed the comment block including
- to tweak the contents to make it fit into the description of the dockerfile better.
- to add the space between remark texts and the leading dial mark.

**Pull request merged**